src/expression.rs

```diff
@@ -584,6 +592,7 @@ impl<F: Field> BinaryExpression<F> {
             |sum, (i, bit)| sum + (&bit.expression * (Element::one() << i)))
     }
 
+    #[cfg(not(feature = "no-std"))]
     pub fn dependencies(&self) -> Vec<Wire> {
         let mut all = HashSet::new();
```

Comment:
Could we use BTreeSet in both cases? I assume performance would be comparable

Reply:
Agreed. I'll change.